### PR TITLE
Improve auth docs

### PR DESCRIPTION
**Pull request by savonarola**
Fixes [EMQX-10968](https://emqx.atlassian.net/browse/EMQX-10968)

### CODE REVIEW

File: en_US/changes/auth-4.4-to-5.1-compatibility.md [hunk 1]
```diff
@@ -70,7 +70,7 @@ authentication {
 ### Built-in Database (Enhanced Authentication)
 
 - SHA1 hashing support used in EMQX 4.4 is no longer available. Use the `algorithm` parameter to choose between `sha512'`and `sha256` algorithms.
-- `iteration_count` can now be configured to specify the calculation times of Hush function (4096 was implicitly used in EMQX 4.4).
+- `iteration_count` can now be configured to specify the calculation times of Hash function (4096 was implicitly used in EMQX 4.4).
```
Comment (Meggielqk) [inline en_US/changes/auth-4.4-to-5.1-compatibility.md]:
```suggestion
- `iteration_count` can now be configured to specify the calculation times of hash function (4096 was implicitly used in EMQX 4.4).
```

**Comment by Meggielqk** [comment]
In our Style Guide, it says "In general, use second-person pronouns (you, your). " and "Use numerals". Refer to https://emqx.atlassian.net/wiki/spaces/TW/pages/517704090/EMQX+Style+Guide, it is not very complete though.

File: en_US/access-control/authn/redis.md [hunk 2]
```diff
@@ -45,37 +45,37 @@ So, we can also configure `cmd` as `HMGET mqtt:${username} password salt is_supe
...
-    - **Iteration Count**: Specify the iteration times; Default: 4096
-    - **Derived Key Length** (optional): Specify the length of the generated password. You can leave this field blank, then the key length will be determined by the pseudorandom function you selected. 
+- **Password Hash**: the hash function for storing passwords in the database. Options include `plain`, `md5`, `sha`, `bcrypt`, and `pbkdf2`. Additional configuration depends on your selected function:
+  - For `plain`, `md5`, `sha`, `sha256`, or `sha512`:
+    - **Salt Position**: Determine how salt (random data) is added to the password. Options are `suffix`, `prefix`, or `disable`. Keep the default value unless you migrate user credentials from external storage into the EMQX built-in database. Note: Set to `disable` if `plain` is selected. 
+  - For `bcrypt`:
+    - **Salt Rounds**: Set the number of applications of the hash function, expressed as 2^Salt Rounds, also known as the "cost factor". The default is `10`, with a range of `5` to `10`. Opting for a higher value enhances security. Note: Increasing the cost factor by 1 doubles the necessary time for authentication.
```
Comment (wivwiv) [inline en_US/access-control/authn/redis.md:74]: In the latest schema, it looks like the Salt Rounds configuration no longer exists
image

Comment (savonarola) [reply en_US/access-control/authn/redis.md:74]: `bcrypt` is an auxiliary schema for the two really used: `bcrypt_rw` and `bcrypt_rw_api`

Comment (wivwiv) [reply en_US/access-control/authn/redis.md:74]: image
I see that the Salt Rounds configuration is also missing from the dashboard, was it removed (for simplicity of configuration)?
If this field is not present on dashbaord, we don't need to inform users about it in the documentation.

**Review by wivwiv — APPROVED** [review]

**Pull request merged**